cranelift/native/src/lib.rs

```diff
@@ -129,6 +129,14 @@ pub fn builder_with_options(infer_native_flags: bool) -> Result<isa::Builder, &'
         if std::arch::is_aarch64_feature_detected!("lse") {
             isa_builder.enable("has_lse").unwrap();
         }
+
+        if std::arch::is_aarch64_feature_detected!("paca") {
+            isa_builder.enable("has_pauth").unwrap();
+        }
+
+        if cfg!(target_os = "macos") {
+            isa_builder.enable("sign_return_address_with_bkey").unwrap();
```

Comment:
Together with the `unimplemented` above, is this going to create issues by default on macOS/aarch64? Should we avoid turning on PAuth on this platform until we implement that?

Reply:
@cfallin My thinking was that since macOS hardcoded the use of the B key, the additional metadata specifying the key was unnecessary, hence this logic was safe. Now that we have our custom unwinder, this issue might be moot anyway, except perhaps for debugging/profiling purposes.

Also, note that back-edge CFI is disabled by default, since `sign_return_address` is the setting that makes the real change. Actually, I had a thought that since pointer authentication was always supported on Apple Silicon, we might treat that platform differently, i.e. by adding a line here:
```
isa_builder.enable("sign_return_address").unwrap();
```

Reply:
Ah, I had missed that it's still off by default. My real concern was just whether we were going to hit `unimplemented` bits immediately on macOS but it sounds like that's not the case.

... and I just tested on my M1 machine and all tests pass, so this indeed is fine.

I'm personally fine with enabling it by default; @bnjbvr do you have any opinions on this, given you did the initial M1 porting work? Flipping the switch can happen in a followup PR if we decide to do so; we don't need to delay this PR I think.

Reply:
As a further clarification, since we are using the `HINT` encodings by default, it should actually be safe to enable the mechanism everywhere (modulo the unwinder issues discussed in #3183, which, as I said before, might be moot by now), but at least on Apple Silicon we have a guarantee that we are not filling up the processor pipeline with no-ops.

Reply:
If this passes all tests with this setting, I don't see any downside in enabling it by default, and only benefits, so I'd be in favor of doing this. Perhaps we'd need some basic documentation explaining that "if you run into this particular kind of SIG* failure, this might be because of pointer authentication", and make sure it's still possible to disable it, would we run into a bug that wasn't caught by the test suite.

Reply:
I can do a candidate PR for this that one of you can test - of course, it is going to pass the CI checks trivially.